### ISSUE TITLE
Validate DataConnectionLink properties [HZ-2314]

### ISSUE THREAD

**Issue by neilstevenson** [hazelcast/hazelcast#24246]
**Please describe the problem you are trying to solve**
If I forget a property on `DataConnectionLink`, eg.
```
        Properties mySqlProperties1 = new Properties();
        //mySqlProperties1.setProperty("data-connection-ref", DATA_CONNECTION_MYSQL);
        mySqlProperties1.setProperty("mapping-type", "JDBC");
        mySqlProperties1.setProperty("table-name", NAME1);
        mySqlProperties1.setProperty("id-column", "hash");
        mySqlProperties1.setProperty("column", "when_ts");
```

then the error from `5.3.0-BETA-2` is unhelpful
```
Caused by: com.hazelcast.core.HazelcastException: Data connection 'null' not found
	at com.hazelcast.dataconnection.impl.DataConnectionServiceImpl.typeForDataConnection(DataConnectionServiceImpl.java:158)
	at com.hazelcast.jet.sql.impl.schema.TableResolverImpl.extractConnector(TableResolverImpl.java:173)
	at com.hazelcast.jet.sql.impl.schema.TableResolverImpl.resolveMapping(TableResolverImpl.java:147)
	at com.hazelcast.jet.sql.impl.schema.TableResolverImpl.createMapping(TableResolverImpl.java:126)
	at com.hazelcast.jet.sql.impl.PlanExecutor.execute(PlanExecutor.java:169)
	at com.hazelcast.jet.sql.impl.SqlPlanImpl$CreateMappingPlan.execute(SqlPlanImpl.java:151)
	at com.hazelcast.sql.impl.SqlServiceImpl.query0(SqlServiceImpl.java:266)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:204)
```

**Please describe the desired behavior**
Validate all required properties are provided

**Describe alternatives you've considered**
None

**Comment by frant-hartm** [comment]
+1 for the title :-D

**Comment by k-jamroz** [comment]
Apparently implemented by #24311
